test/integration/integration.cc

```diff
@@ -918,6 +918,23 @@ void BaseIntegrationTest::testNoHost() {
   EXPECT_TRUE(response.find("HTTP/1.1 400 Bad Request\r\n") == 0);
 }
 
+void BaseIntegrationTest::testAbsolutePath() {
```

Comment:
Also are there tests of failing the new code or were you waiting on sendProtocolError?